plugin/keyring_vault/vault_io.h

```diff
@@ -0,0 +1,54 @@
+#ifndef MYSQL_VAULT_IO_H
+#define MYSQL_VAULT_IO_H
+
+#include <my_global.h>
+#include <logger.h>
+#include "i_vault_io.h"
+#include "vault_parser.h"
+#include "vault_curl.h"
+#include "vault_key_serializer.h"
+
+namespace keyring {
+
+class Vault_io : public IVault_io
+{
+public:
+  Vault_io(ILogger *logger, IVault_curl *vault_curl,
```

Comment:
I will change only logger to reference type, vault_curl and vault_parser are owned by Vault_io and destroyed in Vault_io destructor.